packages/flutter/test/widgets/tap_region_test.dart

```diff
@@ -1053,4 +1053,97 @@ void main() {
     await click(find.text('Outside Surface'));
     expect(tappedInside, isEmpty);
   });
+
+  testWidgets('TapRegion should register and unregister correctly on push and pop', (WidgetTester tester) async {
+    const ValueKey<String> tapRegion1Key = ValueKey<String>('TapRegion');
+    const ValueKey<String> tapRegion2Key = ValueKey<String>('TapRegion2');
+
+    int count1 = 0;
+    int count2 = 0;
+
+    final TapRegion tapRegion1 = TapRegion(
+      key: tapRegion1Key,
+      onTapOutside: (PointerEvent event) {
+        count1++;
+      },
+      child: Container(
+        width: 100,
+        height: 100,
+        color: Colors.red,
+      ),
+    );
+
+    final TapRegion tapRegion2 = TapRegion(
+      key: tapRegion2Key,
+      onTapOutside: (PointerEvent event) {
+        count2++;
+      },
+      child: Container(
+        width: 100,
+        height: 100,
+        color: Colors.red,
+      ),
+    );
+
+    Widget buildPage(Widget body, {FloatingActionButton? fab}) {
+      return Scaffold(
+        appBar: AppBar(
+          backgroundColor: Colors.blue,
+          title: const Text('Test Page'),
+        ),
+        body: Center(child: body),
+        floatingActionButton: fab,
+      );
+    }
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: buildPage(
+          tapRegion1,
+          fab: FloatingActionButton(
+            onPressed: () {
+              Navigator.push(
+                tester.element(find.byType(FloatingActionButton)),
+                MaterialPageRoute<Widget>(builder: (BuildContext context) => buildPage(tapRegion2)),
```

Comment:
nit: void out the type argument

```suggestion
                MaterialPageRoute<void>(builder: (BuildContext context) => buildPage(tapRegion2)),
```

<br>

Depending on whether you decide to keep the `buildPage` method, it could be:

```suggestion
                MaterialPageRoute<void>(
                  builder: (BuildContext context) => Scaffold(
                    body: Center(child: tapRegion2),
                  ),
                ),
```

Reply:
If you're interested in what the route's type argument is used for: it's to allow a "pop with result".
For example, you could show a dialog with yes/no buttons, and one of them could do `Navigator.pop(context, true)` and the other could do `Navigator.pop(context, false)`.

Then you'd use it like so:

```dart
final bool? result = await showDialog<bool>(
  context: context,
  builder: (context) => const YesNoDialog(),
);
```